src/main/resources/avro/metadata.avdl

```diff
@@ -109,4 +109,32 @@ record Dataset {
 
 }
 
+/**
+An analysis sample is a conceptual "sample" for the analysis purpose only.
+Its name appears in a @RG line in a SAM file and in the sample header line of a
+VCF file.
+*/
+record AnalysisSample {
+  /**
+  Identifier of this analysis sample
+  */
+  string id;
+
+  /**
+  The identifier of the dataset this sample belongs to
+  */
+  string datasetId;
+
+  /**
+  Submitted sample name in the dataset. This name uniquely identifies the
+  sample in the dataset.
```

Comment:
It is not just a good practice. It is a basic requirement. We will certainly have a query "give me the genotypes of NA12878". If "NA12878" is not an unambiguous sample label, this query will be impossible. Note that similarly in references.avdl, references in a referenceSet ought to have different names; otherwise it will be very tricky to use references.